swift/Arrow/Sources/ArrowC/include/ArrowCData.h

```diff
@@ -0,0 +1,85 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#ifndef ARROW_C_DATA_INTERFACE
+#define ARROW_C_DATA_INTERFACE
+
+#define ARROW_FLAG_DICTIONARY_ORDERED 1
+#define ARROW_FLAG_NULLABLE 2
+#define ARROW_FLAG_MAP_KEYS_SORTED 4
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>  // Must have this!
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct ArrowSchema {
+    // Array type description
+    const char* format;
+    const char* name;
+    const char* metadata;
+    int64_t flags;
+    int64_t n_children;
+    struct ArrowSchema** children;
+    struct ArrowSchema* dictionary;
+
+    // Release callback
+    void (*release)(struct ArrowSchema*);
+    // Opaque producer-specific data
+    void* private_data;
+};
+
+struct ArrowArray {
+    // Array data description
+    int64_t length;
+    int64_t null_count;
+    int64_t offset;
+    int64_t n_buffers;
+    int64_t n_children;
+    const void** buffers;
+    struct ArrowArray** children;
+    struct ArrowArray* dictionary;
+
+    // Release callback
+    void (*release)(struct ArrowArray*);
+    // Opaque producer-specific data
+    void* private_data;
+};
+
+// Not able to set the release on the schema
+// to NULL in swift.  nil in swift is not 
+// equivalent to NULL.
+void ClearReleaseSchema(struct ArrowSchema*);
+
+// Not able to set the release on the array
+// to NULL in swift.  nil in swift is not 
+// equivalent to NULL.
+void ClearReleaseArray(struct ArrowArray*);
+
+int ArrowSchemaIsReleased(struct ArrowSchema*);
+
+int ArrowArrayIsReleased(struct ArrowArray*);
```

Comment:
How about using a prefix to avoid naming conflict as much as possible?
`ArrowSwiftXXX()`?

(If symbols in `ArrowC` are loaded with `RTLD_LOCAL`, we don't need prefix.)